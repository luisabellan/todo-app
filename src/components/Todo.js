import React from "react";

export default function Todo(props) {
  return (
    <div
      className={`item ${props.item.completed ? " completed" : ""}`}
      onClick={() => props.toggleItem(props.item.id)}
    >
      <p style={{ fontFamily: "Indie Flower", fontSize: "2rem" }}>
        {props.item.name}
      </p>
    </div>
  );
}
