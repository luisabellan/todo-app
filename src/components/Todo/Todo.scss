.task {
  display: flex;
  font-family: "Indie Flower";
  input {
    margin-right: 1rem;
  }
  span {
    font-size: "3rem";
  }
}
