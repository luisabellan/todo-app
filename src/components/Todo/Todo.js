/* eslint-disable no-labels */
/* eslint-disable no-unused-expressions */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Todo.scss';

Todo.propTypes = {
  todo: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    completed: PropTypes.bool,
  }),

  toggleItem: PropTypes.func,
  handleChanges: PropTypes.func
};

export default function Todo(props) {

  const todoInitialState = {
    id: "",
    name: "",
    completed: false
  }

  const [todo, setTodo] = useState(todoInitialState)


  return (
    <div className="checkbox" data-testid="todo">
      <input
        type="checkbox"
        data-testid={`checkbox ${props.todo.name}`}
        name={`item ${props.todo.name}`}
        value={props.todo.name}
        onClick={() => props.toggleItem(props.todo.id)}
        onChange={props.handleChanges}
        defaultChecked={props.todo.completed}
      ></input>
      <span data-testid="todo-output">{props.todo.name}</span>
    </div >
  );
}
