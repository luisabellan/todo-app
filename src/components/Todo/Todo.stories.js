import React from 'react';

import Todo from './Todo';

export default {
  component: Todo,
  title: 'Todo',
};

const Template = args => <Todo {...args} />;

export const Default = Template.bind({});
Default.args = {
  todo: {
    id: new Date(2021, 0, 1, 9, 0),
    name: 'Test Todo',
    completed: false,
  },
};

export const Undone = Template.bind({});
Undone.args = {
  todo: {
    ...Default.args.todo,
    completed: false,
  },
};

export const Done = Template.bind({});
Done.args = {
  todo: {
    ...Default.args.todo,
    completed: true,
  },
};
