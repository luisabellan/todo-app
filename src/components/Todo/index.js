/* eslint-disable no-labels */
/* eslint-disable no-unused-expressions */
import React from "react";
import PropTypes from "prop-types";
import "./Todo.scss";

/* item: PropTypes.exact({
  id: PropTypes.number.isRequired,
  name: PropTypes.string,
  completed: PropTypes.bool,
}); */

Todo.propTypes = {
  item: PropTypes.object,
  toggleItem: PropTypes.func,
  handleChanges: PropTypes.func,
};

export default function Todo(props) {
  return (
    <div
      /* className={`item ${props.item.completed ? " completed" : ""}`}  */
      className="task"
      /*onClick={() => props.toggleItem(props.item.id)}*/
    >
      <input
        type="checkbox"
        name={`item ${props.item.name}`}
        value={props.item.name}
        onClick={() => props.toggleItem(props.item.id)}
        onChange={props.handleChanges}
        defaultChecked={props.item.completed}
      ></input>
      <span>{props.item.name}</span>
    </div>
  );
}
