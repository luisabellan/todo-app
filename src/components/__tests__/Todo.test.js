import React from 'react';
import { v4 as uuidv4 } from 'uuid';

import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { render, screen, cleanup } from '@testing-library/react';
import Todo from '../Todo/Todo';


const todo = {
  id: 'a',
  name: 'buy bread',
  completed: false
};

beforeEach(() => {
  render(<Todo todo={todo} />);

})

describe('todo test', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Todo todo={todo} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show todo', () => {

    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    screen.debug()
    expect(screen.getByTestId('todo-output')).toBeInTheDocument();
  });

  it('click checkbox', () => {
    let checkbox = screen.getByRole('checkbox')
    let output = screen.getByTestId('todo-output')
    expect(checkbox).toBeInTheDocument();
    expect(output).toBeInTheDocument();
    expect(screen.getByTestId('a')).not.toBeChecked();

    userEvent.click(screen.getByTestId('a'));

    expect(screen.getByTestId('a')).toBeChecked();
  });



});
