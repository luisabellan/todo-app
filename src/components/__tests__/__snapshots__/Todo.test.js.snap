// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
    data-testid="todo"
  >
    <input
      data-testid="checkbox"
      name="item buy bread"
      type="checkbox"
      value="buy bread"
    />
    <span
      data-testid="todo-output"
    >
      buy bread
    </span>
  </div>
</DocumentFragment>
`;
